lib/Controller/PollController.php

```diff
@@ -307,12 +381,13 @@ public function closePoll(int $pollId): DataResponse {
 			return new DataResponse(['error' => 'poll'], Http::STATUS_BAD_REQUEST);
 		}
 
-		$poll->setStatus(Poll::STATUS_CLOSED);
-
 		try {
-			$this->pollService->updatePoll($this->participant, $poll);
-		} catch (WrongPermissionsException) {
+			$this->pollService->closePoll($this->participant, $poll);
+		} catch (WrongPermissionsException $e) {
 			return new DataResponse(['error' => 'poll'], Http::STATUS_FORBIDDEN);
+		} catch (Exception $e) {
+			$this->logger->error($e->getMessage(), ['exception' => $e]);
+			return new DataResponse(['error' => 'poll'], Http::STATUS_INTERNAL_SERVER_ERROR);
```

Comment:
```suggestion
```

We should try to avoid 500, as this is way too generic.
Also according to the docs, only `WrongPermissionsException` is thrown, so we don't need to catch and handle anything else. Last resort exception handling is done in the ocs/vX.php including proper logging:
https://github.com/nextcloud/server/blob/9836e9b16484582d309c8437ab46d82e34956941/ocs/v1.php#L61

Reply:
So 403 forbidden?